internal/command/deploy/machines.go

```diff
@@ -185,6 +185,14 @@ func (md *machineDeployment) setMachinesForDeployment(ctx context.Context) error
 		}
 	}
 
+	for _, m := range machines {
+		if m.Config != nil && m.Config.Metadata != nil {
+			if m.Config.Metadata[api.MachineConfigMetadataKeyFlyProcessGroup] == "" {
+				m.Config.Metadata[api.MachineConfigMetadataKeyFlyProcessGroup] = md.appConfig.DefaultProcessName()
```

Comment:
There must be something I am not following because I don’t understand why this is needed.

If the machine's metadata says it is a v2 app machine, I am sure the later .ToMachine("")  call will force set the default process group. Isn't it? 

Maybe that I am missing some intermediary machine grouping by process group.

Reply:
I missed this comment, I merged the PR on my phone 💀

no, this does end up making a difference - maybe it shouldn't, but it does

if you create an app with `fly apps create`, `fly m run` some machines into it, then `fly m update` them into managed machines, their process group is unset, and on the first deploy they would just get deleted because they're considered `""` instead of `"app"`

there's probably something wrong here, for sure